### PR TITLE
5.9 toppings backend

### PR DESCRIPTION
**Pull request by steget**
The admin page now gets the toppings from the backend and you can edit/delete/add new toppings and it gets synced between them all.

### CODE REVIEW

**Comment by amandadehlen** [comment]
"import BootstrapSwitchButton from "bootstrap-switch-button-react";" vad importerar ni för att kunna använda denna?

File: client/src/components/admin/menuItem/topping.js [hunk 1]
```diff
@@ -24,16 +50,167 @@ class Topping extends Component {
...
+
+    //return the price field, either as an input or just a text. Depends on if editMode is active or not
+    getPrice() {
+        if (this.state.editMode) {
+            return (
+                <input
+    className="form-control"
+    type="number"
+    defaultValue={this.state.price}
+    onChange={this.priceChanged}
```
Comment (Minus98) [inline client/src/components/admin/menuItem/topping.js:125]: add
`onKeyPress={this.checkForEnter}`

File: client/src/components/admin/menuItem/topping.js [hunk 2]
```diff
@@ -24,16 +50,167 @@ class Topping extends Component {
...
+
+    //return the name field, either as an input or just a text. Depends on if editMode is active or not
+    getName() {
+        if (this.state.editMode) {
+            return (
+                <input
+                    className="form-control"
+                    type="text"
+                    defaultValue={this.state.name}
+                    onChange={this.nameChanged}
```
Comment (Minus98) [inline client/src/components/admin/menuItem/topping.js:109]: add `onKeyPress={this.checkForEnter}`

File: client/src/components/admin/menuItem/topping.js [hunk 3]
```diff
@@ -1,18 +1,44 @@
...
+            },
         });
+
+        this.notifyBackendOnStockChange = this.notifyBackendOnStockChange.bind(
+            this
+        );
+        this.onToggle = this.onToggle.bind(this);
+        this.save = this.save.bind(this);
+        this.nameChanged = this.nameChanged.bind(this);
+        this.priceChanged = this.priceChanged.bind(this);
```
Comment (Minus98) [inline client/src/components/admin/menuItem/topping.js:35]: and add the following bind 
` this.checkForEnter = this.checkForEnter.bind(this);`

File: client/src/components/admin/menuItem/topping.js [hunk 4]
```diff
@@ -24,16 +51,175 @@ class Topping extends Component {
...
+                            className="btn btn-outline-primary btn-sm"
+                            onClick={() => this.edit()}
+                        >
+                            <img src={edit} alt="edit" />
+                        </button>
+                    </div>
+                </div>
+            );
+        }
+    }
```
Comment (Minus98) [inline client/src/components/admin/menuItem/topping.js:196]: There is **a lot** of duplicated code here, I suggest the following functional decomposition
```suggestion
       renderDeleteButton() {
        if (!this.state.available) {
            return (
                <div className="col-1 px-0">
                    <button
                        className="btn btn-danger btn-sm"
                        type="button"
                        onClick={(e) => this.props.delete(this.state.id, e)}
                    >
                        <img src={trash} alt="remove" />
                    </button>
                </div>
            );
        } else {
            return null;
        }
    }

    getDivs() {
        return (
            <div className="row align-items-center">
                {this.renderDeleteButton()}
                <div className="col-4">{this.getName()}</div>
                <div className="col-3 pr-0">{this.getPrice()}</div>
                <div className="col-2">
                    <BootstrapSwitchButton
                        checked={this.state.inStock}
                        size="sm"
                        onstyle="success"
                        onChange={this.onToggle}
                    />
                </div>
                <div className="col-1">
                    <button
                        type="button"
                        className="btn btn-outline-dark btn-sm"
                        onClick={() => this.edit()}
                    >
                        <img src={edit} alt="edit" />
                    </button>
                </div>
            </div>
        );
    }
```

**Pull request merged**